### PR TITLE
iOS App floor map

### LINKED ISSUE #137: iOS App floor map
## Kind (Required)

- Proposal / Discussion

 ## Overview (Required)

- Since the event venue is large, I think it would be more convenient to have a floor map.

 ## Links	

-

### PR DESCRIPTION
**Pull request by entaku19890818**
## Issue
- close #137

## Overview (Required)
- iOS App floor map

## Links
-

## Screenshot
Before | After
:--: | :--:
nil | 
nil |

### CODE REVIEW

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift [hunk 1]
```diff
@@ -43,13 +43,33 @@ final class SidebarViewController: UITableViewController {
 
         switch switchType {
         case .timeline:
-            rootViewController?.navigationDrawerController?.toggleLeftView()
+            if let _ = UIApplication.topViewController() as? FilterViewController {
```
Comment (ry-itto) [inline ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: Is `rootViewController` property useless?

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
+//
+//  FloorMapViewController.swift
+//  DroidKaigi 2020
+//
+//  Created by 遠藤拓弥 on 2020/01/26.
+//
+
```
[the remark this answers is not in the document]
Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: deleted!

File: ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        menuItem.rx.tap
+            .bind(to: Binder(self) { me, _ in
+                me.navigationDrawerController?.toggleLeftView()
+            }).disposed(by: disposeBag)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
```
[the remark this answers is not in the document]
Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/FloorMap/FloorMapViewController.swift]: removed

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift (hunk 1 above)
Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: It's miss.
Fix reverted

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift [hunk 4]
```diff
@@ -43,13 +54,33 @@ final class SidebarViewController: UITableViewController {
...
+            if let _ = UIApplication.topViewController() as? FloorMapViewController {
+                navigationDrawerController?.toggleLeftView()
+            } else {
+                let vc = FloorMapViewController.instantiate()
+                let nvc = NavigationController(rootViewController: vc)
+                let root = NavigationDrawerController(rootViewController: nvc, leftViewController: SidebarViewController.instantiate(rootViewController: nvc, dependency: .init(switchType: switchType)))
+                navigationDrawerController?.transition(to: root, completion: { _ in
+                    self.navigationDrawerController?.toggleLeftView()
+               })
+            }
```
Comment (ry-itto) [inline ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: In this part, we should match implement way. Please fix like this https://github.com/DroidKaigi/conference-app-2020/pull/597#discussion_r371398392 🙏

Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: OK!
I'll change the code.

**Comment by entaku19890818** [comment]
@ry-itto 
I think fix all issue!

**Comment by ry-itto** [comment]
@entaku19890818 
Thanks! But it looks still conflicting... Please fix it?

**Comment by entaku19890818** [comment]
Fixed it!

**Comment by roana0229** [comment]
Awesome 😄 
Thank you for your dedicated work.

**Comment by ry-itto** [comment]
I think this is not fitting the design...
For example, bar color.

|current|design|
|--|--|
||スクリーンショット 2020-01-31 16 29 45|

**Comment by entaku19890818** [comment]
Sorry....
I will change bar color when leave to  my home.
Are you pointing out that the size of the map is different?

**Comment by ry-itto** [comment]
> Are you pointing out that the size of the map is different?
No, I don't care that. 😄 

Please add view's title and change color 🙏

File: ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift [hunk 5]
```diff
@@ -42,13 +53,28 @@ final class SidebarViewController: UITableViewController {
...
+
+
         case .about:
             break
         case .info:
             break
         case .map:
-            break
+            if let _ = UIApplication.topViewController() as? FloorMapViewController {
```
Comment (ry-itto) [inline ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: After fixing [this issue](https://github.com/DroidKaigi/conference-app-2020/issues/674), we don't need to judge showing `ViewController` with this way. 
Would you fix this code like [this](https://github.com/DroidKaigi/conference-app-2020/pull/672/files#diff-34264a1af610b65a3b561894e675d97eR47)?
```swift
rootViewController.viewControllers.first is AboutViewController
```

And please use `setViewControllers` to move `ViewController` 🙏

Comment (ry-itto) [reply ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: I'm sorry.. I reviewed about this https://github.com/DroidKaigi/conference-app-2020/pull/624#discussion_r371607053 🙇

Comment (ry-itto) [reply ios-base/DroidKaigi 2020/Sidebar/Views/SidebarViewController.swift]: Sorry, you don't have to do about this 🙇

**Comment by entaku19890818** [comment]
@ry-itto 
After merging the master again and make init I got the following error messages.😓
Do you know the cause?

```
MacBook-Pro-2:ios-base endo$ make init
mkdir -p /Users/endo/conference-app-2020/ios-base/DroidKaigi\ 2020/Generated
mint run swiftgen swiftgen
🌱  Using SwiftGen/SwiftGen 6.1.0 from Mintfile.
🌱  Cloning SwiftGen 6.1.0
🌱  Resolving package
🌱  Building package
[1/10] Compiling writer.c
[2/10] Compiling reader.c
[3/10] Compiling parser.c
[4/10] Compiling api.c
[5/10] Compiling emitter.c
[6/10] Compiling scanner.c
[7/11] Compiling Kanna CSS.swift
/private/var/folders/tc/7k760n257395s_tvy3xdhm1m0000gq/T/mint/github.com_SwiftGen_SwiftGen/.build/checkouts/Kanna/Modules/libxml2-kanna.h:1:9: note: while building module 'libxml2' imported from /private/var/folders/tc/7k760n257395s_tvy3xdhm1m0000gq/T/mint/github.com_SwiftGen_SwiftGen/.build/checkouts/Kanna/Modules/libxml2-kanna.h:1:
#import <libxml2/libxml/HTMLtree.h>
        ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "libxml/HTMLparser.h"
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/libxml2/libxml/HTMLparser.h:15:10: error: 'libxml/xmlversion.h' file not found
#include <libxml/xmlversion.h>
         ^
<module-includes>:1:9: note: in file included from <module-includes>:1:
#import "libxml2-kanna.h"
        ^
/private/var/folders/tc/7k760n257395s_tvy3xdhm1m0000gq/T/mint/github.com_SwiftGen_SwiftGen/.build/checkouts/Kanna/Modules/libxml2-kanna.h:1:9: error: could not build module 'libxml2'
#import <libxml2/libxml/HTMLtree.h>
        ^
/private/var/folders/tc/7k760n257395s_tvy3xdhm1m0000gq/T/mint/github.com_SwiftGen_SwiftGen/.build/checkouts/Kanna/Sources/Kanna/CSS.swift:27:8: error: could not build Objective-C module 'libxmlKanna'
import libxmlKanna
       ^
'SwiftClibxml2' /private/var/folders/tc/7k760n257395s_tvy3xdhm1m0000gq/T/mint/github.com_SwiftGen_SwiftGen/.build/checkouts/SwiftClibxml2: warning: system packages are deprecated; use system library targets instead
🌱  Encountered error during "swift build -c release -Xswiftc -target -Xswiftc x86_64-apple-macosx10.15". Use --verbose to see full output
🌱  Failed to build SwiftGen 6.1.0 with SPM
make: *** [swiftgen] Error 1
```

**Comment by entaku19890818** [comment]
https://github.com/DroidKaigi/conference-app-2020/issues/667

**Comment by ry-itto** [comment]
Oh sorry.. Please install `libxml2` by `brew` 🙏

**Comment by entaku19890818** [comment]
thanks!

File: ios-base/DroidKaigi 2020/en.lproj/Localizable.strings [hunk 6]
```diff
@@ -4,7 +4,7 @@
 "DroidKaigi Description" = "DroidKaigi is a conference tailored for Android developers.\nIt's scheduled to take place on the 20 and 21 of February 2020.";
 "Access" = "Access";
 "Toc Building" = "Toc Building";
-"Maps" = "Maps";
+"Maps" = "Floor Maps";
```
Comment (ry-itto) [inline ios-base/DroidKaigi 2020/en.lproj/Localizable.strings]: This is for `About`'s map strings.
This string represents such as `GoogleMap`.

Please create new Key 🙏

Comment (roana0229) [reply ios-base/DroidKaigi 2020/en.lproj/Localizable.strings]: @entaku19890818  If you have any questions please tell me. 🙏

Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/en.lproj/Localizable.strings]: I had a misunderstanding!
I'll make a new key.

Comment (entaku19890818) [reply ios-base/DroidKaigi 2020/en.lproj/Localizable.strings]: fix done!

**Pull request merged**